Sources/app.js

```diff
@@ -65,11 +67,24 @@ var action = {
             this.onWillAppear(jsn);
         else
             api_request.sendRequest();
+    },
 
-    }
+    onKeyDown: function (jsn) {
+        log('onKeyDown(): ', jsn);
 
+        const api_request = this.cache[jsn.context];
+
+        if(!api_request)
+            this.onWillAppear(jsn);
+        else
+            api_request.sendRequest(false, KEY_TYPES.DOWN);
+    },
 };
 
+const KEY_TYPES = {
+    DOWN: 'DOWN',
+    UP: 'UP',
+};
```

Comment:
I'd prefer if a structure like this was only used when it's needed - given `UP` is the only value actually need (in the switch statement).

Instead, a simple boolean with a default might be more appropriate given my preference that this feature is treated as an extra option, rather than a required setting. I'm wording that badly I suspect - open to thoughts/suggestions.